test_getaddrinfo.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include <assert.h>
+
+#define ASSERT_EQUAL(str1, str2) do { \
+ assert(str1 != NULL); \
+ assert(str2 != NULL); \
+ assert(strcmp(str1, str2)==0); } while (0)
+#define ASSERT assert
+
+void resolve(const char *hostname, char *out) {
+	struct addrinfo hints;
+	struct addrinfo *result;
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = AF_UNSPEC;    /* Allow IPv4 or IPv6 */
+	hints.ai_socktype = SOCK_DGRAM; /* Datagram socket */
+	hints.ai_flags = 0;
+	hints.ai_protocol = 0;          /* Any protocol */
+
+	int s = getaddrinfo(hostname, NULL, &hints, &result);
+	if (s != 0) {
+		out[0] = '\0';
+		return;
+	}
+	if (result == NULL) {
+		out[0] = '\0';
+		return;
+	}
+
+	struct sockaddr_in *addr;
+	addr = (struct sockaddr_in *)result->ai_addr;
```

Comment:
Why assume IPv4 here?

Reply:
The example I looked up didn't use `inet_ntop`. I now migrated to use `inet_ntop`, which does support ipv6 :+1: Added ipv6 tests as well.